src/logion-chain/LogionClientContext.tsx

```diff
@@ -140,21 +141,18 @@ export default function LogionClientContextProvider(props: Props) {
 
 const SIGN_AND_SEND_STRATEGY = new DefaultSignAndSendStrategy();
 
-function getEndpoints(): string[] {
-    const providerSocket = config.PROVIDER_SOCKET;
-    if (providerSocket !== undefined) {
-        return [ providerSocket ];
-    } else {
-        return config.edgeNodes.map(node => node.socket);
-    }
-}
-
 async function connectToLogionWithMetamask(): Promise<LogionClient> {
 
-    const anonymousClient = await LogionClient.create({
-        rpcEndpoints: getEndpoints(),
-        directoryEndpoint: config.directory
-    });
+    let anonymousClient: LogionClient;
+    if (config.environment) {
+        anonymousClient = await newLogionClient(config.environment);
+    } else {
+        anonymousClient = await LogionClient.create({
+            rpcEndpoints: config.rpcEndpoints,
+            directoryEndpoint: config.directory,
+            buildFileUploader: () => new BrowserAxiosFileUploader(),
+        });
+    }
```

Comment:
The above condition is found in almost all our clients, maybe be should move this logic to `newLogionClient`? For example, add a LOCAL environment which expects the endpoints and directory to be passed as an argument?

Reply:
Thanks for the suggestion - created [an issue](https://github.com/logion-network/logion-internal/issues/1170) for that.